### PR TITLE
qa/workunits/cephadm/test_cephadm.sh: move osd test to ceph-volume

### PR DESCRIPTION
**Pull request by mgfritch**
instead of running via the orchestrator

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/workunits/cephadm/test_cephadm.sh [hunk 1]
```diff
@@ -250,11 +250,33 @@ loop_dev=$($SUDO losetup -f)
...
+            lvm list --format json $device_name > $TMPDIR/osd.map
+    osd_id=$($SUDO cat $TMPDIR/osd.map | jq -cr '.. | ."ceph.osd_id"? | select(.)')
+    osd_fsid=$($SUDO cat $TMPDIR/osd.map | jq -cr '.. | ."ceph.osd_fsid"? | select(.)')
+
+    # deploy the osd
+    $CEPHADM deploy --name osd.$osd_id \
+          --fsid $FSID \
+          --keyring $TMPDIR/keyring.bootstrap.osd \
+          --config $CONFIG \
+          --osd-fsid $osd_fsid
```
Comment (sebastian-philipp) [inline qa/workunits/cephadm/test_cephadm.sh]: why? Is the current code broken in some way? this reduces the coverage in mgr/cephadm

Comment (mgfritch) [reply qa/workunits/cephadm/test_cephadm.sh]: `ceph orch osd create` was broken in the `mgr/cephadm` layer  ... shouldn't these tests only focus on `bin/cephadm` ?

Comment (sebastian-philipp) [reply qa/workunits/cephadm/test_cephadm.sh]: hm. do you have a traceback some something similar?

Comment (mgfritch) [reply qa/workunits/cephadm/test_cephadm.sh]:
```
+ sudo /tmp/tmp.ZlyRDRVwla/cephadm --image docker.io/ceph/daemon-base:latest-master-devel shell --fsid 00000000-0000-0000-0000-0000deadbeef --config /tmp/tmp.CJGxN6SA7u/tmp.r9LWfuuEb5 --keyring /tmp/tmp.CJGxN6SA7u/tmp.WY52Ho5uw3 -- ceph orch osd create host1:/dev/test_cephadm/test_cephadm.0
Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1070, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 191, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/usr/share/ceph/mgr/mgr_module.py", line 309, in call
    return self.func(mgr, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/module.py", line 458, in _create_osd
    raise_if_exception(completion)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 701, in raise_if_exception
    raise e
TypeError: %d format: a number is required, not str
```

Comment (sebastian-philipp) [reply qa/workunits/cephadm/test_cephadm.sh]: mgr logs should show more!

**Comment by sebastian-philipp** [comment]
I think we need to retarget this to master. cc @tchaikov

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-03-20_13:25:17-rados-wip-kefu-testing-2020-03-20-1836-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-21_04:20:53-rados-wip-kefu-testing-2020-03-21-1031-distro-basic-smithi/

**Pull request merged**